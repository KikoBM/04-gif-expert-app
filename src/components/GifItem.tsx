import { Tooltip } from '@mui/material';
import { useState } from 'react';

interface GifItemProps {
    title: string,
    url: string
}

export const GifItem = ({ title, url }: GifItemProps) => {

    const [copied, setCopied] = useState(false);
    const [showTooltip, setShowTooltip] = useState(false)

    const handleCopy = async () => {
        try {
            await navigator.clipboard.writeText(url);
            setCopied(true)
            setShowTooltip(true)
            setTimeout(() => { setShowTooltip(false) }, 1200)
            setTimeout(() => { setCopied(false) }, 1200)
        } catch (error) {
            console.error('Error al copiar al portapeles', error)
        }
    }

    const handleMouseEnter = () => {
        if (!copied) {
            setShowTooltip(true)
            setTimeout(() => setShowTooltip(false), 1500)
        }
    }
    const handleMouseLeave = () => {
        if (!copied) {
            setShowTooltip(false)
        }
    }

    return (
        <Tooltip title={copied ? 'Â¡Copiado!' : 'Copiar link'} open={showTooltip} arrow slotProps={{
            transition: { timeout: 0 }
        }}>
            <div className="card" onClick={handleCopy} onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
                <img src={url} alt={title} />
                <p>{title}</p>
            </div>
        </Tooltip>
    )
}


