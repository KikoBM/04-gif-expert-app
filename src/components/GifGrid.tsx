
import { GifItem } from "./GifItem"
import { useFetchGifs } from "../hooks/useFetchGifs"


interface GifGridProps {
    category: string
}

export const GifGrid = ({ category }: GifGridProps) => {

    const { images, isLoading } = useFetchGifs(category)
    console.log(images, isLoading)


    return (
        <div className="category-container">
            <h3> {category} </h3>

            <div className="card-grid">
                {
                    images.map(img => (
                        <GifItem
                            key={img.id}
                            { ...img }
                        />
                    ))
                }
            </div>
        </div>
    )
}
