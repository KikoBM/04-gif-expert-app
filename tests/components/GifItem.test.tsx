import { render, screen } from '@testing-library/react'
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en GifItem.tsx', () => {

    const title = 'Saitama'
    const url = "https://one-punch.com/saitama.jpg"

    test('debe hacer match con el snapshot ', () => {
        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })

    test('debe mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem title={title}  url={url}/>)

        const {src, alt} = screen.getByRole('img') as HTMLImageElement
        expect(src).toBe(url)
        expect(alt).toBe(title)
    })
    


})
