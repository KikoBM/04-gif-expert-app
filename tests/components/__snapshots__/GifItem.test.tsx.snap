// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en GifItem.tsx debe hacer match con el snapshot  1`] = `
<div>
  <div
    aria-label="Copiar link"
    class="card"
    data-mui-internal-clone-element="true"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
